### ISSUE TITLE
Long Term Support of Container App Backing Resources for Purposes of Private Endpoint Connectivity

### ISSUE THREAD

**Issue by hoffmanntravis** [microsoft/azure-container-apps#600]
Hello,

We are looking to secure our container apps via some sort of private connectivity mechanism.  Our initial attempt used a load balancer with Azure Private Link Service attached to the same vnet as our container apps (we supply a vnet) as well as, in theory, a Private Endpoint.  However, that fell apart since Private Link Service does not support load balancers with an ip address as the backend (as opposed to a NIC that corresponds to a VM or VMSS, issue [here](https://github.com/MicrosoftDocs/azure-docs/issues/75134)).

We then set up an app gateway with private endpoint, that worked pretty well but is overly messy and somewhat expensive for our needs.

Finally, we setup a private link service and a private endpoint pointed directly at the AKS internal load balancer that backs container app resources (backing resources described [here](https://learn.microsoft.com/en-us/azure/container-apps/networking#managed-resources)).  This works extremely well.  We can find the corresponding resource with infrastructure as code since it is tagged appropriately (it seems to transparently do this).  The setup is easy and straightforward.

However, my colleague is concerned that support for targeting these backing resources via private link service could be pulled at any time.  What I would like to know is do you have any roadmap plans to hide the backing resources or modify them in some way that this method would break, or can we continue to rely on this mechanism to provide private endpoint connectivity?

This [link](https://dev.to/kaiwalter/preliminary-private-linking-an-azure-container-app-environment-3cnf) provides a full description of what we setup if it's not clear what I am talking about.

Thanks!!
Travis

**Comment by hoffmanntravis** [comment]
Any update here?  We are getting close to making key decisions around production implementations with regard to this feature.

**Comment by SophCarp** [comment]
Hi @hoffmanntravis thank you for reaching out! We know private endpoint connectivity is a pain point for many users, and I'm glad you have a successful work around. We don't have any short term plans to remove support for your set up.  

Long term, we plan on improving private endpoints. That change will be communicated ahead of time in our roadmap. Let me know if you have any questions.

**Comment by hoffmanntravis** [comment]
Thanks for the reply @SophCarp .  We will continue building as planned then and will also keep an eye out for further updates on this topic.  If Microsoft can add Private Link Service support for non VM backends connected to a load balancer that would work as well ([ticket](https://github.com/MicrosoftDocs/azure-docs/issues/75134)), but PE availability on container apps as you described is the ideal solution.  A single PE endpoint connected to the container environment load balancer like we are doing with more official support is a worthwhile consideration also so the PE can be set up once for all containers associated with the container environment, but I can see advantages to having discrete PEs for each individual container app.  Ultimately we can work with any of the above really.

For anyone following along here, this is how we retrieved the load balancer in PowerShell:
`$kubernetesLoadBalancer = (Get-AzResource -Tag @{"managed environment name" = "<container environment name>" } -ResourceType "Microsoft.Network/loadBalancers" -Name "kubernetes-internal")`

Note that this approach comes with all of the caveats associated with this thread and the core question I posed.

Feel free to close this from my perspective unless you are using it to track feature implementation or need it open for your own usage.
